packages/daemon/src/app/controller/plugin.ts

```diff
@@ -48,4 +47,35 @@ export class PluginController {
     });
     successRes(this.ctx, { data: plugins });
   }
+
+  @post('/upload')
+  public async upload() {
+    const fs = await this.ctx.getFileStream();
+    const id = await this.pluginManager.installFromTarStream(fs);
+    successRes(this.ctx, { id });
+  }
+
+  @get('/log')
+  public async log() {
+    const logStream = await this.pluginManager.getInstallLogStream(this.ctx.query.id);
+    const sse = new ServerSentEmitter(this.ctx);
+    if (logStream) {
+      await new Promise(() => {
```

Comment:
No resolve function is called which causes https://github.com/alibaba/pipcook/pull/342/files#r454101633.